### ISSUE TITLE
Closure of OpenWeather's One Call 2.5 API?

### ISSUE THREAD

**Issue by mattblovell** [Bodmer/OpenWeather#24]
Any update plans given the (apparent) closure of OpenWeather's One Call API 2.5 occurring in June 2024?

**How to transfer from One Call API 2.5 to the One Call API 3.0**
[https://openweathermap.org/one-call-transfer](https://openweathermap.org/one-call-transfer)

I don't immediately see anything that significant between API 2.5 and API 3.0.  Perhaps it is as simple as updating the root of the URL that gets composed?

**Comment by wabbitguy** [comment]
@mattblovell I just tried that simple one call with the 2.5 changed to 3.0 and got an error message. I have a subscription (free) but there's more changes they've made obviously. So something they don't like about the URL.

`{"cod":401, "message": "Please note that using One Call 3.0 requires a separate subscription to the One Call by Call plan. Learn more here https://openweathermap.org/price. If you have a valid subscription to the One Call by Call plan, but still receive this error, then please see https://openweathermap.org/faq#error401 for more info."}`

**Comment by mattblovell** [comment]
I believe one has to explicitly _add_ One Call 3.0 access, even for an existing key.  (Note that you can place a self-imposed limit on the number of One Call 3.0 API invocations that get processed, such that you definitely stay below the threshold where charges start.)

I went ahead and added One Call 3.0 to my existing account (and key, presumably), but I've not yet tried accessing it.

**Comment by wabbitguy** [comment]
@mattblovell thanks for that nugget, I just read someone else had said the same thing, that even if you had a subscription, you had to resubscribe to get 3.0 access. Perhaps that's their way of cleaning out all the unused 2.5 accounts in their database.

Did you resubscribe or simply add a new API key in your existing account? I didn't find any concise info on adding a OneCall 3.0 to my account, or my eyes are too blurry from reading all the stuff...sigh...

**Comment by mattblovell** [comment]
> Did you resubscribe or simply add a new API key in your existing account?

I just added the new API to my existing account.  

I just have the one pre-existing key but, as noted, I've noted yet tried any accesses, so I don't know if that will work or if a new key is needed.  I think OpenWeather provides a nice way of trying accesses, so I guess that will be my next step.

**Comment by wabbitguy** [comment]
@mattblovell thanks! I tried that first with the 2.5 change to 3.0 and the new API key but I wondered if there was a propagation delay with enabling the new API key. Going to try it again...Sure make it easy don't they..LOL

Update: just added a new API key, tried to use it with the older 2.5 URL and it said it didn't know me. Tried it with the 3.0 URL and got the same error. Going to leave it in the system for a while and see if it updates.

Just checked and their web site says after 2hrs the key will activate.

**Comment by mattblovell** [comment]
I added the 3.0 API access several days ago.  Trying "raw" in Firefox just now with the representative example of

`https://api.openweathermap.org/data/3.0/onecall?lat={lat}&lon={lon}&exclude={part}&appid=[{API key}]`

it appears to have worked!  I'll try the modifying my little weather widget itself over the weekend.

**Comment by wabbitguy** [comment]
@mattblovell awesome! I was trying the Safari URL for access the same as you did and got the error. So now I have added an API key and I have also resubscribed with a different email. Which was interesting because the confirming email that just showed up has the 2.5 URL in it...had me laughing a little...

Going to wait a few hours and see what havoc it bestows on me. I've become dependent on this little weather station...:-)

**Comment by mattblovell** [comment]
I was able to indeed have the existing OpenWeather code work fine just changing "2.5" to "3.0" within the URL.

It took a while to get to that state, since the desktop I'm working on did not have the Arduino IDE installed.  Grabbing the latest one, there were different hoops to jump through than previously to get LittleFS support working.  I also did not immediately notice the change in the most recent libraries that require passing the `LittleFS` handle to `loadFont()` calls!  Once I got all of that straightened out, though, everything still worked with the 2.5 -> 3.0 transition.

**Comment by wabbitguy** [comment]
@mattblovell after much reading, I subscribed with a credit card and then waited 12 hrs.

Changed the 2.5 to 3.0 and it worked!

I'm using the older Arduino 1.8.9 IDE (cause it complains less about stuff than the new "improved one"..LOL), ESP8266 and 2.8" TFT. I've modified the original display code a fair bit so the main display also rotates between the moon/sunrise/etc and a 6 hr forecast for POP%, Temp and Dew Point.

I also coded in BigDataCloud to turn long/lat into locations for the display. Added a switch so I can toggle between two locations.

Like you, I had to change the SPIFFS to LittleFS but that was easier than I'd thought it was going to be so fairly painless.

Be nice if there was a way to see usage on the OpenWeatherOneCall site but darned if I can find any "counter"...

Anyway, thanks for the help, hints and great we got it working!

![Monosnap 2024-04-27 19-54-03](https://github.com/Bodmer/OpenWeather/assets/8953419/690b3433-2fea-412d-8101-411804df0cec)

**Comment by RonAdex** [comment]
Since October 15, my weather update has stopped working, even though I've upgraded to API 3.0.

How is it with you? Do I need to change the key to a new one?
Are there any other changes needed?
![20241016_191343_001](https://github.com/user-attachments/assets/ee6e2d10-755e-48b5-9639-f24eb2331afb)

**Comment by wabbitguy** [comment]
@RonAdex I already had subscribed to OpenWeather with a credit card and they give you 1,000 calls a day and then charge your card if you go over that. Once that's done, and it takes a few hours for it to go through their system..I changed the URL that gets the weather (OpenWeather.cpp line 53) for the 3.0 API:

`  String url = "https://api.openweathermap.org/data/3.0/onecall?lat=" + latitude + "&lon=" + longitude + "&exclude=minutely," + exclude + "&units=" + units + "&lang=" + language + "&appid=" + api_key;`

I didn't generate a new key, just used the default one.

**Comment by RonAdex** [comment]
> I didn't generate a new key, just used the default one.

Yes, I actually noticed that in the OpenWeather.cpp file there is an address  with the numbering 2.5 in two places   ;)

Thank you for your response!

**Comment by wabbitguy** [comment]
> > I didn't generate a new key, just used the default one.
> 
> Yes, I actually noticed that in the OpenWeather.cpp file there is an address with the numbering 2.5 in two places ;)
> 
> Thank you for your response!

@RonAdex in my OpenWeather.cpp I only had the 2.5 on the single line (#53). So that's the one I changed to 3.0 to make it work (if I remember right, it was a few months back)...
